src/PrestaShopBundle/EventListener/ActionDispatcherLegacyHooksSubscriber.php

```diff
@@ -79,7 +79,10 @@ public function callActionDispatcherBeforeHook(FilterControllerEvent $event)
 
         $requestAttributes = $event->getRequest()->attributes;
         $controllerType = self::NA_CONTROLLER;
-        $controller = $event->getController()[0];
+        $controller = is_array($event->getController())
+            ? $event->getController()[0]
+            : $event->getController()
+        ;
```

Comment:
As said by @lartist in my PR, "we don't have control on key or index" so we can use reset like this :

```suggestion
        $controller = $event->getController();

        if (is_array($controller)) {
            $controller = reset($controller);
        }
```

Reply:
I'm not sure it's worth it here, getController() will return a callable so we know for sure that if it's an array, the part we're looking for is at index `0`.

Reply:
I think @lartist say that we are sure if It's an array but we are not sure that there is numeric or string indexes